jumphost/ssh.go

```diff
@@ -31,18 +31,32 @@ type SshTunnel struct {
 	ctx       context.Context
 }
 
-func NewSshClient(hostname, username, password string, port int) SshClient {
-	authenticationMethods := []ssh.AuthMethod{
-		ssh.Password(password),
+func NewSshClient(hostname, username, password string, keyPair string, useAgent bool, port int) SshClient {
+	authenticationMethods := make([]ssh.AuthMethod, 0)
+	if password != "" {
+		authenticationMethods = append(authenticationMethods, ssh.Password(password))
 	}
-	socket := os.Getenv("SSH_AUTH_SOCK")
-	conn, err := net.Dial("unix", socket)
-	if err != nil {
-		log.Printf("failed to connect to ssh agent %v", err)
-	} else {
-		agentClient := agent.NewClient(conn)
-		authenticationMethods = append(authenticationMethods, ssh.PublicKeysCallback(agentClient.Signers))
+
+	if useAgent {
+		agentConn, err := net.Dial("unix", os.Getenv("SSH_AUTH_SOCK"))
+		if err != nil {
+			log.Printf("Failed to connect to ssh-agent: %s", err)
```

Comment:
```suggestion
			log.Printf("failed to connect to ssh-agent: %s", err)
```
iirc this is a suggestion on go lint to not start errors with capital case